activestate.yaml

```diff
@@ -33,9 +33,77 @@ scripts:
 
       Type `${A}state run learn${R}` to view a quick start guide of commonly used commands.
       EOT
+  - name: convert
+    description: Bring user to a webpage to convert their anonymous work to a project
+    language: perl
+    value: |
+      use Browser::Open qw( open_browser );
+      $proj = "${project}";
+      $proj =~ m/\/commit\/(.+)$/;
+      my $commit = $1;
+      my $preview_url = "https://platform.activestate.com/project-preview/${commit}";
+      print <<EOT;
+      If your browser doesn't open automatically, you can visit: ${preview_url}
+      EOT
+      open_browser($preview_url);
+  - name: pkg-before
+    description: Message before packages install to inform users about build status on the webpage
+    language: perl
+    value: |
+      use if $^O eq "MSWin32", Win32::Console::ANSI;
+      my $H = "\e[1;39;1m"; # Highlight
+      my $A = "\e[1;36m"; # Actionable
+      my $R = "\e[0m"; # Reset
+      my $proj = "${project}";
+      $proj =~ m/\/commit\/(.+)$/;
+      my $commit = $1;
+      my $preview_url = "https://platform.activestate.com/project-preview/${commit}";
+      my $msg = "";
+      my $note = "";
+      my $user = "${mixin.user.name}";
+      if ($user eq "") {$note = "${H}Note:${R} If you sign-up for a platform account you can monitor build progress and receive an email when it completes, as well as other advanced project management functionality."; $msg = "Visit ${A}${preview_url}${R} to sign-up for account to see in-progress build-status.";}
+      
+      if ($note ne "" and $msg ne "") {
+      print <<EOT;
+      
+      ${note}
+      
+      ${msg}
+
+      EOT
+      }
+  - name: pkg-after
+    description: Message after packages install to inform users about build status on the webpage
+    language: perl
+    value: |
+      use if $^O eq "MSWin32", Win32::Console::ANSI;
+      my $H = "\e[1;39;1m"; # Highlight
+      my $A = "\e[1;36m"; # Actionable
+      my $R = "\e[0m"; # Reset
+      my $anon=0;
+      my $proj = "${project}";
+      $proj =~ m/\/commit\/(.+)$/;
+      my $commit = $1;
+      my $preview_url = "https://platform.activestate.com/project-preview/${commit}";
+      my $msg = "";
+      my $user = "${mixin.user.name}";
+      if ($user eq "") {$msg = "${H}Note:${R} You're currently using this tool anonymously.\nSign-up for an account on the Activestate Platform to create your own version of this project.\nHaving an account gives you: full customization, team sharing, versioned backup, and more!\n\nVisit: ${A}${preview_url}${R} or run ${A}state run convert${R}."}
+      if ($msg ne "") {
+      print <<EOT;
+      
+      ${msg}
+      EOT
+      }
+
```

Comment:
This is getting very complex and noisy for the activestate.yaml. Suggest moving this into a .pl file and just using the activestate.yaml for controller logic.

Reply:
I'm on board with this. I only put it in the yaml because someone (you?) suggested we put stuff in the yaml! I'm all for that — it will mean I can share logic too.

Reply:
A change in the dashboard made this way simpler so I basically just deleted a bunch of code which made it not really necessary to have a separate perl file.